lnrpc/routerrpc/forward_interceptor.go

```diff
@@ -213,15 +212,26 @@ func (r *forwardInterceptor) resolveFromClient(
 // the store. Before they are removed it ensure to resume as the default
 // behavior.
 func (r *forwardInterceptor) onDisconnect() {
-	// Then close the channel so all go routine will exit.
-	close(r.quit)
 
-	log.Infof("RPC interceptor disconnected, resolving held packets")
-	for key, forward := range r.holdForwards {
-		if err := forward.Resume(); err != nil {
-			log.Errorf("failed to resume hold forward %v", err)
+	if !r.server.cfg.AlwaysIntercept {
+		log.Infof("RPC interceptor disconnected, resolving held packets")
+		for key, forward := range r.holdForwards {
+			if err := forward.Resume(); err != nil {
```

Comment:
Does it need to be done in this PR or can it be a follow up?

Reply:
Oh yeah, we had some trouble with this in production against ACINQs node due to having a few edge cases in our height watching logic. It would be nice to move it back down into LND. I'm not familiar with the work on #5206 so I'm not sure I can tackle that now. Given that this PR doesn't quite make things worse in this regard, I think it can be addressed in a follow up.

Reply:
I've created a PR that takes a first step in this direction: https://github.com/lightningnetwork/lnd/pull/6212